### PR TITLE
Add rpc call to get current addresses for our node

### PR DESCRIPTION
**Pull request by michael-loki**
Builds on top of #856 

Also, add the ability to use libcurl to hit the jsonrpc endpoint (the libcurl dependency is optional, if it's not available on the system, support simply isn't compiled in).

### CODE REVIEW

File: daemon/rcutil.cpp [hunk 1]
```diff
@@ -55,26 +122,31 @@ main(int argc, char* argv[])
 
   options.add_options()
       ("v,verbose", "Verbose", cxxopts::value<bool>())
-      ("c,colour", "colour output", cxxopts::value<bool>()->default_value("true"))
       ("h,help", "help", cxxopts::value<bool>())
-      ("j,json", "output in json", cxxopts::value<bool>())
```
Comment (michael-loki) [inline daemon/rcutil.cpp]: based on feedback from @jagerman, the non-json output isn't that useful, so make it the default.

File: llarp/config/config.cpp [hunk 2]
```diff
@@ -475,6 +475,23 @@ namespace llarp
...
   }
 
+  fs::path
+  GetDefaultConfigDir()
+  {
+#ifdef _WIN32
+    fs::path homedir = fs::path(getenv("APPDATA"));
+#else
+    fs::path homedir = fs::path(getenv("HOME"));
+#endif
```
Comment (majestrate) [inline llarp/config/config.cpp]: can these be `const` ?

Comment (michael-loki) [reply llarp/config/config.cpp]: yes

**Comment by majestrate** [comment]
looks like ci wants the package: `libcurl4-openssl-dev` not `libcurl-dev`

**Comment by necro-nemesis** [comment]
Glad to see this.

**Pull request merged**